### PR TITLE
do not allocate gauge if cardinality metrics are turned off

### PR DESCRIPTION
**Pull request by brawndou**
Changes:
- Cached gauges will not be allocated if omitCardinalityMetrics is set to true

### CODE REVIEW

**Comment by LukasJenicek** [comment]
@brawndou 

Thanks for the fast fix :+1:. This was causing a panic in our go-chi telemetry middleware and I could not turn off the cardinality metrics easily.

```
panic: descriptor Desc{fqName: "tally.internal.counter_cardinality", help: "tally.internal.counter_cardinality gauge", constLabels: {}, variableLabels: {version,host,instance}} is invalid: "tally.internal.counter_cardinality" is not a valid metric name

goroutine 1 [running]:
github.com/uber-go/tally/v4/prometheus.NewReporter.func1({0x9be240, 0xc000226000})
	/home/lukas-jenicek/Projects/Go-Chi/telemetry/vendor/github.com/uber-go/tally/v4/prometheus/reporter.go:306 +0x5e
github.com/uber-go/tally/v4/prometheus.(*reporter).AllocateGauge(0xc00003a080, {0x91411d, 0x22}, 0xc0000a4390)
	/home/lukas-jenicek/Projects/Go-Chi/telemetry/vendor/github.com/uber-go/tally/v4/prometheus/reporter.go:415 +0x194
github.com/uber-go/tally/v4.newScopeRegistryWithShardCount(0xc000002180, 0xc0000a4240?, 0x0, 0x5?)
	/home/lukas-jenicek/Projects/Go-Chi/telemetry/vendor/github.com/uber-go/tally/v4/scope_registry.go:107 +0x654
github.com/uber-go/tally/v4.newRootScope({0xc0000a4240, {0x904b71, 0x4}, {0x0, 0x0}, {0x7f7b643f9008, 0xc00003a080}, {0x9bb9a8, 0x1}, {0x9c2e08, ...}, ...}, ...)
	/home/lukas-jenicek/Projects/Go-Chi/telemetry/vendor/github.com/uber-go/tally/v4/scope.go:192 +0x79b
github.com/uber-go/tally/v4.NewRootScope(...)
	/home/lukas-jenicek/Projects/Go-Chi/telemetry/vendor/github.com/uber-go/tally/v4/scope.go:118
github.com/go-chi/telemetry.newRootScope({0x0, {0x904b71, 0x4}, {0x0, 0x0}, {0x7f7b643f9008, 0xc00003a080}, {0x9bb9a8, 0x1}, {0x0, ...}, ...}, ...)
```

Do you know why the separator of the metrics name was changed from underscore to dot  in the latest version v1.41.0 ?

File: scope_registry.go [hunk 1]
```diff
@@ -103,7 +103,7 @@ func newScopeRegistryWithShardCount(
 		}
 		r.subscopes[i].s[scopeRegistryKey(root.prefix, root.tags)] = root
 	}
-	if r.root.cachedReporter != nil {
+	if r.root.cachedReporter != nil && !omitCardinalityMetrics {
```
Comment (SokolAndrey) [inline scope_registry.go:106]: pls add unit test for this

Comment (brawndou) [reply scope_registry.go:106]: done

File: scope_registry_test.go [hunk 2]
```diff
@@ -221,3 +221,22 @@ func TestForEachScopeConcurrent(t *testing.T) {
 
 	<-done
 }
+
+func TestCachedReporterNoAllocOnOmitInternalMetrics(t *testing.T) {
+	r := newTestStatsReporter()
+	root, closer := NewRootScope(ScopeOptions{CachedReporter: r, OmitCardinalityMetrics: true}, 0)
```
Comment (SokolAndrey) [inline scope_registry_test.go]: do you want to check that it gets allocated when `OmitCardinalityMetrics` is false?

**Review by SokolAndrey — APPROVED** [review]
lgtm

**Pull request merged**